File: fields/bindings/java/examples/field_vd_t.java

//
// Copyright (c) 2013 Limit Point Systems, Inc.
//

// Test driver for field_vd.java

//$$SCRIBBLE: This is just a placeholder.  It will be
//            further developed as part of the field algebra project.

import bindings.java.*;

public class field_vd_t extends base_example
{
  static final String moduleName = "fields_java_binding";
  static final String className  = "field_vd_t";

  public field_vd_t()
  {
    super(moduleName);
  }

  public field_vd_t(String module_name)
  {
    super(module_name);
  }

  public static void usage()
  {
    String nl = System.getProperty("line.separator");

    StringBuffer sb = new StringBuffer();
    sb.append("Usage: java [-cp CLASSPATH] " + className
              + " [-option]").append(nl)
    .append(nl)
    .append("where options include:").append(nl)
    .append("    i_size (default = 1)").append(nl)
    .append("    -help  Print this message and exit.").append(nl)
    .append("    -h     Print this message and exit.").append(nl)
    .append(nl);

    System.out.println(sb);
    System.exit(1);
  }

  public static void main(String argv[])
  {
    // Handle command line arguments.

    if(argv.length > 0)
    {
      usage();
    }

    // Instantiate "this" class (loads the libraries, etc.).

    field_vd_t test = new field_vd_t();

    // Make the fiber_bundles namespace.

    fiber_bundles_namespace lns = new fiber_bundles_namespace(className);

    // Output a text version to stdout.

    //System.out.println(the_namespace.to_string());

    // Write namespace to file.

    test.writeHdfFile(lns, className +".hdf");

  }
}

File: fields/bindings/python/examples/field_vd_t.py
#! /usr/bin/env python
#
# Copyright (c) 2013 Limit Point Systems, Inc. 
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

##$$SCRIBBLE: This is just a placeholder.  It will be
##            further developed as part of the field algebra project.

##$$TODO: Figure out how to combine these.

from sheaves_python_binding import *
from fiber_bundles_python_binding import *
from fields_python_binding import *

## Make the sheaves namespace.

ns = fiber_bundles_namespace("field_vd_t")

## Output a text version to stdout.

ns.to_stream()

##  Need read access to write namespace to a file.

ns.get_read_write_access()

##  Write it to a file.

sa = storage_agent("field_vd_t.hdf")
sa.write_entire(ns)

